lxd/auth/types.go

```diff
@@ -48,6 +48,9 @@ type Authorizer interface {
 	//
 	// Warn: You almost never need this function. You should use GetPermissionChecker instead.
 	GetPermissionCheckerWithoutEffectiveProject(ctx context.Context, entitlement Entitlement, entityType entity.Type) (PermissionChecker, error)
+
+	// GetViewableProjects accepts a list of permissions and returns a list of projects that a member of a group with these permissions is able to view.
+	GetViewableProjects(ctx context.Context, permissions []api.Permission) ([]string, error)
```

Comment:
Fell at the first hurdle :)

> accepts a list of permissions and returns a list of projects that a member of a group with these permissions is able to view.

I found this pretty cryptic and couldn't reason about it after several attempts.

What permissions around I pass into this?

Reply:
> What permissions around I pass into this?

I don't follow sorry.

In the API, only groups can be assigned permissions. Identities are granted access only via the groups they are a member of. Given a list of permissions for a group, this function returns a list of project names that a member of only that group would be able to see.